output/transform/postgres_types.go

```diff
@@ -0,0 +1,43 @@
+package transform
+
+import (
+  snapshot "github.com/pganalyze/collector/output/pganalyze_collector"
+  "github.com/pganalyze/collector/state"
+)
+
+func transformPostgresTypes(s snapshot.FullSnapshot, transientState state.TransientState, databaseOidToIdx OidToIdx) (snapshot.FullSnapshot, OidToIdx) {
+  typeOidToIdx := make(OidToIdx)
+
+  for _, pgType := range transientState.Types {
+    var customType snapshot.CustomTypeInformation_Type
+    switch pgType.Type {
+    case "e":
+      customType = snapshot.CustomTypeInformation_ENUM
+    case "d":
+      customType = snapshot.CustomTypeInformation_DOMAIN
+    case "c":
+      customType = snapshot.CustomTypeInformation_COMPOSITE
```

Comment:
Thinking through this again, I think we either need to:

(1) Add all custom type types here (i.e. missing would be `p` for a pseudo-type, `r` for a range type, or `m` for a multirange type)

(2) Filter out these other types in the input query, like we filter out `b` today

The benefit of (1) would be that we know how many people are using range types, to determine whether its worth supporting it. But not sure if its worth the effort.

Reply:
Tracking all custom types makes sense, like you said, so we have a better idea what types our customers are using. I'll also update it to store `b` types.

Reply:
Ah, one thing to note about the base types is that e.g. btree_gist types get captured.

<img width="439" alt="Screen Shot 2021-10-03 at 9 52 37 AM" src="https://user-images.githubusercontent.com/688886/135759459-482ea426-a671-4c98-9bb6-99fea5b8aaa3.png">

An earlier version of this PR filtered out types that aren't used by columns, but I removed that to speed up the query. They could be filtered by the Go code since at that point we have the columns, but that would mean looping over the columns an extra time.

Thoughts?

Reply:
I went ahead and kept base types excluded from the query for now.